src/fsharp/FSharp.Core/async.fsi

```diff
@@ -435,6 +432,112 @@ namespace Microsoft.FSharp.Control
             computation:Async<'T> * ?cancellationToken:CancellationToken-> Task<'T>
 
 
+    /// <summary>The F# compiler emits references to this type to implement F# async expressions.</summary>
+    type AsyncReturn
+
+    /// <summary>The F# compiler emits references to this type to implement F# async expressions.</summary>
+    [<Struct; NoEquality; NoComparison>]
+    type AsyncActivation<'T> =
```

Comment:
Should we add some attribute here like `CompilerGenerated` or `Obsolete` or something else to indicate users should not use the type directly?